dts/bindings/i3c/i3c-device.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2017, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for I3C devices
+
+include: [base.yaml, power.yaml]
+
+on-bus: i3c
+
+properties:
+    reg:
+      required: true
+      description: |
+        Contains 3 fields.
+
+        For I3C devices, the 3 fields are static address, first half
+        of Provisioned ID, and the second half of the Provisioned ID.
+          1. The static address can be assigned to be the target device's
+             dynamic address before Dynamic Address Assignment (DAA)
+             starts. Can be zero if static address is not used, and
+             target address is determined via DAA.
+          2. First half of the Provisioned ID contains the manufacturer
+             ID left-shifted by 1, where the manufacturer ID is
+             the bits 33-47 (zero-based) of the 48-bit Provisioned ID.
+          3. Second half of the Provisioned ID contains the combination of
+             the part ID (bits 16-31 of the Provisioned ID) left-shifted
+             by 16, and the instance ID (bits 12-15 of the Provisioned ID)
+             left-shifted by 12. Basically, this is the lower 32 bits
+             of the Provisioned ID.
```

Comment:
The Provisioned ID, AFAIK, does not have a dedicated field to indicate that. The last 12 bits are vendor specific. From what I have read the spec, the "initial dynamic address" is the static address of the device at power up. The controller is free to assign another dynamic address if the target device supports those commands.

Reply:
Sorry, the line put this comment on made this confusing... I wasn't referring to anything with it's PID. What I was referring to here was that should there be an additional field such as `init-dyn-addr:`.

Also, from the spec... there is no "initial dynamic address" until assignment. It will still respond to i2c calls to it's static address but won't respond to i3c calls... but the dyanmic address assigned with SETDASA doesn't have to be the same as it's static address. Linux handles it in the same way where if the board definition has an `init-dyn-addr`, then it will 'attempt' to assign that address with SETDASA if requested. If there isn't an `init-dyn-addr`, then it will just reuse it's static address as it's dynamic address.
https://github.com/torvalds/linux/blob/74164d284b2909de0ba13518cc063e9ea9334749/include/linux/i3c/master.h#L160